actors/cache/impl/zcache.go

```diff
@@ -0,0 +1,211 @@
+			Local:              cacheConfig.Local,
+			Remote:             cacheConfig.Remote,
+		}
+
+		shortRobustMapConfig := &zcache.CombinedConfig{
+			GlobalPrefix:       fmt.Sprintf("%s%s", prefix, Short2RobustMapPrefix),
+			GlobalTtlSeconds:   -1,
+			IsRemoteBestEffort: cacheConfig.IsRemoteBestEffort,
+			Local:              cacheConfig.Local,
+			Remote:             cacheConfig.Remote,
+		}
+
+		if m.shortCidMap, err = zcache.NewCombinedCache(shortCidMapConfig); err != nil {
+			return err
+		}
+		if m.robustShortMap, err = zcache.NewCombinedCache(robustShortMapConfig); err != nil {
+			return err
+		}
+		if m.shortRobustMap, err = zcache.NewCombinedCache(shortRobustMapConfig); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (m *ZCache) ImplementationType() string {
+	return ZCacheImpl + "/" + m.cacheType
+}
+
+func (m *ZCache) BackFill() error {
+	// Nothing to do
+	return nil
+}
+
+func (m *ZCache) GetActorCode(address address.Address, key filTypes.TipSetKey) (string, error) {
+	shortAddress, err := m.GetShortAddress(address)
+	if err != nil {
+		m.logger.Sugar().Infof("[ActorsCache] - Error getting short address: %s\n", err.Error())
+		return cid.Undef.String(), common.ErrKeyNotFound
+	}
+
+	var code string
+	ctx := context.Background()
+	if err = m.shortCidMap.Get(ctx, shortAddress, &code); err != nil {
+		return cid.Undef.String(), common.ErrKeyNotFound
+	}
+
+	if code == "" {
+		return cid.Undef.String(), common.ErrEmptyValue
+	}
+
+	return code, nil
+}
+
+func (m *ZCache) GetRobustAddress(address address.Address) (string, error) {
+	isRobustAddress, err := common.IsRobustAddress(address)
+	if err != nil {
+		return "", err
+	}
+
+	if isRobustAddress {
+		// Already a robust address
+		return address.String(), nil
+	}
+
+	// This is a short address, get the robust one
+	var robustAdd string
+	ctx := context.Background()
+	if err = m.shortRobustMap.Get(ctx, address.String(), &robustAdd); err != nil {
+		return "", common.ErrKeyNotFound
+	}
+
+	if robustAdd == "" {
+		return "", common.ErrEmptyValue
+	}
+
+	return robustAdd, nil
+}
+
+func (m *ZCache) GetShortAddress(address address.Address) (string, error) {
+	isRobustAddress, err := common.IsRobustAddress(address)
+	if err != nil {
+		return "", err
+	}
+
+	if !isRobustAddress {
+		// Already a short address
+		return address.String(), nil
+	}
+
+	// This is a robust address, get the short one
+	var shortAdd string
+	ctx := context.Background()
+
+	if err = m.robustShortMap.Get(ctx, address.String(), &shortAdd); err != nil {
+		return "", common.ErrKeyNotFound
+	}
+
+	if shortAdd == "" {
+		return "", common.ErrEmptyValue
+	}
+
+	return shortAdd, nil
+}
+
+func (m *ZCache) storeRobustShort(robust string, short string) {
+	if robust == "" || short == "" {
+		// zap.S().Warn("[ActorsCache] - Trying to store empty robust or short address")
+		return
+	}
+
+	ctx := context.Background()
+	_ = m.robustShortMap.Set(ctx, robust, short, -1)
```

Comment:
It's unclear what -1 signifies. We could use a named constant to make this more readable (and in the other parts that -1 appears)